src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx

```diff
@@ -74,94 +75,135 @@ const ConfirmationPage = ({
   );
   return (
     <div className="confirmation-page">
-      <p className="vads-u-font-weight--bold print-copy">
-        Please print this page for your records.
-      </p>
-      <AlertBox
-        headline="Your order has been submitted"
-        className="order-submission-alert"
-        content={
-          <p>
-            We'll send you an email confirming your order to{' '}
-            <strong>{vetEmail}</strong>.
+      {selectedProductArray.length > 0 && (
+        <>
+          <p className="vads-u-font-weight--bold print-copy">
+            Please print this page for your records.
           </p>
-        }
-        status="success"
-      />
-      <AlertBox
-        className="order-summary-alert"
-        content={
-          <section>
-            <h4 className="vads-u-margin-top--0">
-              Request for Batteries and Accessories{' '}
-              <span className="vads-u-font-weight--normal">(Form 2346A)</span>
-            </h4>
-            <p className="vads-u-margin--0">
-              for {fullName.first} {fullName.last}
-            </p>
-            <p className="vads-u-margin-bottom--0">
-              <strong>Items ordered</strong>
+          <AlertBox
+            headline="Your order has been submitted"
+            className="order-submission-alert"
+            content={
+              <p>
+                We'll send you an email confirming your order to{' '}
+                <strong>{vetEmail}</strong>.
+              </p>
+            }
+            status="success"
+          />
+          <AlertBox
+            className="order-summary-alert"
+            content={
+              <section>
+                <h4 className="vads-u-margin-top--0">
+                  Request for Batteries and Accessories{' '}
+                  <span className="vads-u-font-weight--normal">
+                    (Form 2346A)
+                  </span>
+                </h4>
+                <p className="vads-u-margin--0">
+                  for {fullName.first} {fullName.last}
+                </p>
+                <p className="vads-u-margin-bottom--0">
+                  <strong>Items ordered</strong>
+                </p>
+                <ul className="vads-u-margin-bottom--1">
+                  {selectedProductArray.map(product => (
+                    <li key={product.productId}>
```

Comment:
Do you think its worth while to add optional chaining to nullsafe all of these objects we're accessing?

Reply:
Yeah, I agree with extra protection for our objects!